pkg/unittest/test_job.go

```diff
@@ -208,7 +208,11 @@ func (t *TestJob) RunV3(
 	// Replace --- with ---\n to ensure yaml rendering is parsed correctly/
 	// Still questioning why rendering in helm template is working, as it is using similar steps.
 	for file, rendered := range outputOfFiles {
```

Comment:
Hello @ivankatliarchuk 

The splitterPattern regex can be set at the start of the parseYamlFile method (line 81)

Once set, we can remove the whole for loop, as the parseYamlFile is only called when it is an yaml file and before assertion happens.

This will avoid double logic like to determine if a file is a yaml file.

Greetings,
@quintush

Reply:
Moved to line 81